dGame/dGameMessages/GameMessages.cpp

```diff
@@ -6204,3 +6205,18 @@ void GameMessages::SendShowBillboardInteractIcon(const SystemAddress& sysAddr, L
 	if (sysAddr == UNASSIGNED_SYSTEM_ADDRESS) SEND_PACKET_BROADCAST
 	else SEND_PACKET
 }
+
+void GameMessages::HandleRequestActivityExit(RakNet::BitStream* inStream, Entity* entity) {
+	bool canceled = false;
+	inStream->Read(canceled);
+	if (!canceled) return;
+
+	LWOOBJID player_id = LWOOBJID_EMPTY;
+	inStream->Read(player_id);
+	auto player = EntityManager::Instance()->GetEntity(player_id);
+	if (!entity || !player) return;
+
+	for (auto* shootingGallery : EntityManager::Instance()->GetEntitiesByComponent(eReplicaComponentType::SHOOTING_GALLERY)) {
```

Comment:
This GM is not sent by _any_ other activity?  Could you confirm this?

Reply:
The only other place I saw it being sent was in roller component, which we don't have implemented, but I'll double check

Reply:
I'd double check integration testing just in case imo.  Not all GMs are sent from the client exe, some are sent straight from lua, straight from flash etc and the GM call would not be visible in the client (assuming you just looked for references to the function in the client and not in flash/lua)

Reply:
Lua scripts only used it in sg and old racing